### ISSUE TITLE
Use NMC lifetime model from Smith 2017

### ISSUE THREAD

**Issue by dguittet** [NREL/SAM#577]
The NMC lifetime model from Smith 2017 has been implemented in SSC and requires adjustments to the GUI so that the model can be selected and used.

**Comment by brtietz** [comment]
@dguittet - I think this is close to being closed, the only thing remaining might be the upgrade script. A model saved in 2020.11.29 will default to the NMC life model when opened in develop, regardless of the chemistry. It might be simplest to have the upgrade script default to cycle-life for this release, but I could also see the argument for a chemistry based upgrade path.

**Comment by cpaulgilman** [comment]
This should be addressed in https://github.com/NREL/SAM/pull/714.

Probably best to test this in Develop after #714 is merged because of all of the UI form updates in the pull request.

**Comment by brtietz** [comment]
I tested this today, and it's close but still a little buggy. The correct life option is selected on the life page, but one can't see the cycle life chart without toggling off of that option and back on:

![image](https://user-images.githubusercontent.com/5530592/140584466-4f777661-9f0d-4913-912f-ec9ecbb33977.png)

**Comment by cpaulgilman** [comment]
That's on me. I thought we wanted to hide the Cycle Degradation and Calendar Degradation tables for the Li-ion NMC/Graphite and Li-ion LMO/LTO battery lifetime options, and only show the tables for the Cycle and Calendar Degradation option to prevent users from inadvertently changing parameters, so I designed it to work that way.

**Comment by cpaulgilman** [comment]
Clarification from today's SAM meeting: The state in the screenshot above is after opening a file from a previous version, so an issue with versions.lk.

**Comment by cpaulgilman** [comment]
Solution is to set battery life exclusive variable that controls what UI forms appear on Battery Lifetime page in versions.lk. I will do this as part of battery dispatch UI fixes.

Alternative (and better) solution would be to set exclusive var value in on_load function for Battery Life Options form, but this causes SAM to crash.

Snip from versions.lk

```
    if ( is_batt_tech ) {
		value('ui_batt_life_model',0);
		value('batt_life_model',0);
		value('batt_life_excl', value('ui_batt_life_model')); // add this to fix issue reported in #577
		message('The battery replacement and lifetime inputs are now on the new Battery Life input page with a new option to choose a battery life model for different battery types. Be sure to check battery life model option on the Battery Life page when the file opens.','warning');
	}
```

**Comment by cpaulgilman** [comment]
This should be fixed with https://github.com/NREL/SAM/pull/754.
